earth_enterprise/src/server/geserver

```diff
@@ -84,15 +86,61 @@ stop() {
   echo
   echo $dashes
   # only stop if already running...
-  if [ -f /var/opt/google/pgsql/data/postmaster.pid ] ; then
+  if [ -f $pidfile_postmaster ] ; then
     run_as_user $GEPGUSER "$gebin_root/pg_ctl -D /var/opt/google/pgsql/data -l /var/opt/google/pgsql/logs/pg.log stop";
-    rm -f /var/opt/google/pgsql/data/postmaster.pid
+    rm -f $pidfile_postmaster
   fi
   echo $dashes
   echo 'Done Stopping Google Earth Enterprise Server'
   echo
 }
 
+status_postmaster() {
+  if [ -s $pidfile_postmaster ] ; then
+    pid_postmaster=`head -1 $pidfile_postmaster`
+    # redirect to /dev/null because pg_ctl --silent option doesnt work for status subcommand
+    run_as_user $GEPGUSER "$gebin_root/pg_ctl -D /var/opt/google/pgsql/data status" >& /dev/null
+    returncode=$?
+  else
+    returncode=1
+  fi
+  if [ $returncode -eq 0 ]; then
+    echo "postmaster: OK"
```

Comment:
Minor: To make 'running' and 'not running' messages more uniform, and provide more details, I'd use something like:

```BASH
  echo "postmaster:   Running (PID $pid_postmaster)."
```

Reply:
Totally better, yes.